src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,229 @@
+       * need to constrain the width
+       */
+      const positionBottom = (e: HTMLElement): void => {
+        e.style.marginLeft = `${adjustHorizontalCenter}px`
+        if (!isElementInViewport(e)) {
+          e.classList.add('usa-tooltip__body--wrap')
+        }
+        e.style.marginTop = `${adjustToEdgeY + offsetForTopMargin}px`
+      }
+      /**
+       * Positions tooltip at the right
+       */
+      const positionRight = (e: HTMLElement): void => {
+        e.style.marginBottom = '0'
+        e.style.marginLeft = `${adjustToEdgeX + leftOffset}px`
+        e.style.bottom = `${
+          (tooltipHeight - offsetForTooltipBodyHeight) / 2 +
+          offsetForBottomMargin
+        }px`
+      }
+
+      /**
+       * Positions tooltip at the left
+       * @param {HTMLElement} e - this is the tooltip body
+       */
+      const positionLeft = (e: HTMLElement): void => {
+        e.style.marginBottom = '0'
+        if (leftOffset > tooltipBodyWidth) {
+          e.style.marginLeft = `${
+            leftOffset - tooltipBodyWidth - (TRIANGLE_SIZE + SPACER)
+          }px`
+        } else {
+          e.style.marginLeft = `-${
+            tooltipBodyWidth - leftOffset + (TRIANGLE_SIZE + SPACER)
+          }px`
+        }
+        e.style.bottom = `${
+          (tooltipHeight - offsetForTooltipBodyHeight) / 2 +
+          offsetForBottomMargin
+        }px`
+      }
+
+      /**
+       * We try to set the position based on the
+       * original intention, but make adjustments
+       * if the element is clipped out of the viewport
+       */
+      switch (position) {
+        case 'top':
+          positionTop(tooltipBody)
+          if (!isElementInViewport(tooltipBody)) {
+            positionBottom(tooltipBody)
+          }
+          break
+        case 'bottom':
+          positionBottom(tooltipBody)
+          if (!isElementInViewport(tooltipBody)) {
+            positionTop(tooltipBody)
+          }
+          break
+        case 'right':
+          positionRight(tooltipBody)
+          if (!isElementInViewport(tooltipBody)) {
+            positionLeft(tooltipBody)
+            if (!isElementInViewport(tooltipBody)) {
+              positionTop(tooltipBody)
+            }
+          }
+          break
+        case 'left':
+          positionLeft(tooltipBody)
+          if (!isElementInViewport(tooltipBody)) {
+            positionRight(tooltipBody)
+            if (!isElementInViewport(tooltipBody)) {
+              positionTop(tooltipBody)
+            }
+          }
+          break
+
+        default:
+          // skip default case
+          break
+      }
+    }
+  }, [isVisible])
+
+  return (
+    <span
+      data-testid="tooltip"
+      ref={wrapperRef}
+      className="usa-tooltip"
+      {...spanProps}
+      onMouseEnter={() => {
+        setVisible(true)
+      }}
+      onMouseLeave={() => {
+        setVisible(false)
+      }}>
```

Comment:
I just realized these (in addition to the focus triggers etc that Suz mentioned) should be on the child (trigger) element. Right now, if the child element has a huge margin, the tooltip will trigger on mousover of the child element's margin as well, because the span encapsulates that as well.

Reply:
yep im seeing this today